packages/markdown/remark/src/rehype-images.ts

```diff
@@ -22,6 +22,21 @@ export function rehypeImages() {
 
 						node.properties['__ASTRO_IMAGE_'] = JSON.stringify({ ...props, index });
 
+						Object.keys(props).forEach((prop) => {
+							delete node.properties[prop];
+						});
+					} else if (file.data.astro?.remoteImagePaths?.includes(node.properties.src)) {
+						const { ...props } = node.properties;
+
+						const index = imageOccurrenceMap.get(node.properties.src) || 0;
+						imageOccurrenceMap.set(node.properties.src, index + 1);
+
+						node.properties['__ASTRO_IMAGE_'] = JSON.stringify({
+							inferSize: 'width' in props && 'height' in props ? undefined : true,
+							...props,
+							index,
+						});
+
```

Comment:
This new `else if` branch has code that is repeated to the existing `if` branch, can we find a better way to not have repeated code?